modules/apps/portal-language/portal-language-override-service/src/main/java/com/liferay/portal/language/override/internal/util/PLOLanguageOverrideProviderUtil.java

```diff
@@ -0,0 +1,113 @@
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.language.override.internal.util;
+
+import com.liferay.petra.concurrent.DCLSingleton;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.service.CompanyLocalServiceUtil;
+import com.liferay.portal.language.override.model.PLOEntry;
+import com.liferay.portal.language.override.service.PLOEntryLocalServiceUtil;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Supplier;
+
+/**
+ * @author Regisson Cesar
+ */
+public class PLOLanguageOverrideProviderUtil {
+
+	public static void add(PLOEntry ploEntry) {
+		_add(_ploEntriesMapDCLSingleton.getSingleton(_supplier), ploEntry);
+	}
+
+	public static String encodeKey(long companyId, String languageId) {
+		return StringBundler.concat(companyId, StringPool.POUND, languageId);
+	}
+
+	public static Supplier<Map<String, HashMap<String, String>>> getSupplier() {
+		return _supplier;
+	}
+
+	public static void remove(PLOEntry ploEntry) {
+		Map<String, HashMap<String, String>> ploEntriesMap =
+			_ploEntriesMapDCLSingleton.getSingleton(_supplier);
+
+		ploEntriesMap.computeIfPresent(
+			encodeKey(ploEntry.getCompanyId(), ploEntry.getLanguageId()),
+			(key, value) -> {
+				value.remove(ploEntry.getKey());
+
+				if (value.isEmpty()) {
+					return null;
+				}
+
+				return value;
+			});
+	}
+
+	public static void update(PLOEntry ploEntry) {
+		Map<String, HashMap<String, String>> ploEntriesMap =
+			_ploEntriesMapDCLSingleton.getSingleton(_supplier);
+
+		ploEntriesMap.computeIfPresent(
+			encodeKey(ploEntry.getCompanyId(), ploEntry.getLanguageId()),
+			(key, value) -> {
+				value.put(ploEntry.getKey(), ploEntry.getValue());
+
+				return value;
+			});
+	}
+
+	private static void _add(
+		Map<String, HashMap<String, String>> ploEntriesMap, PLOEntry ploEntry) {
+
+		ploEntriesMap.compute(
+			encodeKey(ploEntry.getCompanyId(), ploEntry.getLanguageId()),
+			(key, value) -> {
+				if (value == null) {
+					value = new HashMap<>();
+				}
+
+				value.put(ploEntry.getKey(), ploEntry.getValue());
+
+				return value;
+			});
+	}
+
+	private static Map<String, HashMap<String, String>> _createPLOEntriesMap() {
+		Map<String, HashMap<String, String>> ploEntriesMap =
+			new ConcurrentHashMap<>();
+
+		CompanyLocalServiceUtil.forEachCompanyId(
+			companyId -> {
```

Comment:
Hey @regiCesar21, please use the `Snapshot` approach here, see the `com.liferay.blogs.web.internal.sharing.BlogsEntrySharingUtil` as an example. Thx